scheduler-bindings/src/lib.rs

```diff
@@ -248,26 +248,42 @@ pub struct PackToWorkerMessage {
 
 pub mod pack_message_flags {
     //! Flags for [`crate::PackToWorkerMessage::flags`].
-    //! These flags can be ORed together so must be unique bits, with
-    //! the exception of [`NONE`].
-    //! The *default* behavior, [`NONE`], is to attempt execution and
-    //! inclusion in the specified `max_execution_slot`.
-
-    /// No special handling - execute the transactions normally.
-    pub const NONE: u16 = 0;
-
-    /// Transactions on the [`super::PackToWorkerMessage`] should have their
-    /// addresses resolved.
-    ///
-    /// If this flag, the transaction will attempt to be executed and included
-    /// in the current block.
-    pub const RESOLVE: u16 = 1 << 1;
+    //! Use [`CHECK`] or [`EXECUTE`] to specify how a batch should be processed.
+    //! See [`check_flags`] and [`execution_flags`] for details.
+
+    /// Combine with [`check_flags`] for performing checks on transactions.
+    /// Worker will respond with [`super::worker_message_types::CheckResponse`] if
+    /// the message is processed.
+    pub const CHECK: u16 = 0;
+    /// Combine with additional [`execution_flags`] for executing a batch of transactions.
+    /// Worker will responsd with [`super::worker_message_types::ExecutionResponse`] if
+    /// the message is processed.
+    pub const EXECUTE: u16 = 1;
+
+    pub mod execution_flags {}
+
+    pub mod check_flags {
+        /// Transactions should check status: if transaction has already been processed
+        /// or the nonce is invalid.
+        pub const STATUS_CHECKS: u16 = 1 << 1;
+
+        /// Fee-payer balance should be fetched for transactions.
+        pub const GET_FEE_PAYER_BALANCE: u16 = 1 << 2;
+
+        /// Transactions should have ATL pubkeys resolved and returned.
+        pub const RESOLVE_PUBKEYS: u16 = 1 << 3;
+    }
 }
 
-/// The message was not processed.
-pub const NOT_PROCESSED: u8 = 0;
-/// The message was processed.
-pub const PROCESSED: u8 = 1;
+pub mod processed_codes {
+    /// The message was processed.
+    pub const MESSAGE_PROCESSED: u8 = 0;
+    /// The message was not processed because the message was invalid.
+    pub const INVALID_MESSAGE: u8 = 1;
```

Comment:
Could consider dropping message prefix/postfix as all the variants relate to messages.

Reply:
[23ae290](https://github.com/anza-xyz/agave/pull/8821/commits/23ae2904f7fdd876d181f10ed1b37950c83301f7)